translib/transformer/xfmr_intf.go

```diff
@@ -0,0 +1,2152 @@
+	var err error
+	if inParams.oper == DELETE || inParams.oper == REPLACE {
+		requestUriPath := (NewPathInfo(inParams.requestUri)).YangPath
+		if log.V(3) {
+			log.Info("intf_pre_xfmr:- Request URI path = ", requestUriPath)
+		}
+		errStr := "Delete operation not supported for this path - "
+		if inParams.oper == REPLACE {
+			errStr = "Replace operation not supported for this path - "
+		}
+		switch requestUriPath {
+		case "/openconfig-interfaces:interfaces":
+			errStr += requestUriPath
+			return tlerr.InvalidArgsError{Format: errStr}
+		case "/openconfig-interfaces:interfaces/interface":
+			pathInfo := NewPathInfo(inParams.uri)
+			if len(pathInfo.Vars) == 0 {
+				errStr += requestUriPath
+				return tlerr.InvalidArgsError{Format: errStr}
+			}
+		}
+	}
+	return err
+}
+
+var intf_subintfs_table_xfmr TableXfmrFunc = func(inParams XfmrParams) ([]string, error) {
+	var tblList []string
+
+	pathInfo := NewPathInfo(inParams.uri)
+	idx := pathInfo.Var("index")
+
+	if idx == "" {
+		if inParams.oper == GET || inParams.oper == DELETE {
+			if inParams.dbDataMap != nil {
+				(*inParams.dbDataMap)[db.ConfigDB]["SUBINTF_TBL"] = make(map[string]db.Value)
+				(*inParams.dbDataMap)[db.ConfigDB]["SUBINTF_TBL"]["0"] = db.Value{Field: make(map[string]string)}
+				tblList = append(tblList, "SUBINTF_TBL")
+			}
+		}
+		log.Info("intf_subintfs_table_xfmr - Subinterface get operation ")
+	} else {
+		if idx == "0" {
+			if inParams.dbDataMap != nil {
+				(*inParams.dbDataMap)[db.ConfigDB]["SUBINTF_TBL"] = make(map[string]db.Value)
+				(*inParams.dbDataMap)[db.ConfigDB]["SUBINTF_TBL"]["0"] = db.Value{Field: make(map[string]string)}
+				(*inParams.dbDataMap)[db.ConfigDB]["SUBINTF_TBL"]["0"].Field["NULL"] = "NULL"
+			}
+			tblList = append(tblList, "SUBINTF_TBL")
+		}
+		if log.V(3) {
+			log.Info("intf_subintfs_table_xfmr - Subinterface get operation ")
+		}
+	}
+
+	return tblList, nil
+}
+
+var YangToDb_intf_subintfs_xfmr KeyXfmrYangToDb = func(inParams XfmrParams) (string, error) {
+	var subintf_key string
+	var err error
+
+	log.Info("YangToDb_intf_subintfs_xfmr - inParams.uri ", inParams.uri)
+
+	pathInfo := NewPathInfo(inParams.uri)
+	ifName := pathInfo.Var("name")
+	if ifName == "*" {
+		return ifName, nil
+	}
+
+	intfType, _, ierr := getIntfTypeByName(ifName)
+	if intfType == IntfTypeUnset || ierr != nil {
+		return ifName, errors.New("Invalid interface type IntfTypeUnset")
+	}
+
+	idx := pathInfo.Var("index")
+
+	if idx != "0" && idx != "*" && idx != "" {
+		subintf_key = ifName + "." + idx
+	} else { /* For get 0 index case & subscribe index * case */
+		subintf_key = idx
+	}
+
+	log.Info("YangToDb_intf_subintfs_xfmr - return subintf_key ", subintf_key)
+	return subintf_key, err
+}
+
+var DbToYang_intf_subintfs_xfmr KeyXfmrDbToYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+
+	if log.V(3) {
+		log.Info("Entering DbToYang_intf_subintfs_xfmr")
+	}
+	var idx string
+
+	if strings.Contains(inParams.key, ".") {
```

Comment:
As per HLD, we do not support subinterfaces

Reply:
Updated this.